downloads/downloader.go

```diff
@@ -167,27 +161,23 @@ func (d Downloader) getImageDownloads(ctx context.Context, p Parameters) (Model,
 		return downloads, err
 	}
 
-	available := make(map[string]map[string]Info)
-	for ext, extVal := range image.Downloads {
-		available[ext] = make(map[string]Info)
-		for variant, variantVal := range extVal {
-			available[ext][variant] = Info{
-				URL:     variantVal.Href,
-				Size:    strconv.Itoa(variantVal.Size),
-				Public:  variantVal.Public,
-				Private: variantVal.Private,
-				Skipped: false,
-			}
+	// map image downloads to map of Info structs, assumign Href is the public URL when the image is published or completed
+	available := make(map[string]Info)
+	for k, v := range image.Downloads {
+		available[k] = Info{
+			Public:  v.Href,
+			Private: v.Private,
 		}
 	}
 
+	// The image will be considered published (available for public downloads), when it is in 'published' or 'completed' state.
 	return Model{
-		IsPublished: "published" == image.State,
+		IsPublished: ("published" == image.State || "completed" == image.State),
```

Comment:
Why is the 'completed' state considered published?

Reply:
We defined the `published` state that is set after the image has been moved to the static bucket:
 https://docs.google.com/document/d/1eWUp0L1j5erswFR9xduDjNuqaCg-LII8P3AEJFSRjfk/edit#
So download service should use the public link in this case.